From 93b871ff608a3ac4d7a7973e2acd86f13f29715b Mon Sep 17 00:00:00 2001
From: Carsten Hauge <carsten.hauge@sonymobile.com>
Date: Mon, 7 Jul 2014 09:36:07 +0200
Subject: [PATCH 22/30] Avoid NullPointerException when updating preference
 intents

Comparisons changed on the intent action. Sometimes an account
preference intent does not have an action specified.

Fixes: 67075850
Test: manual - go to Settings > Users & accounts and tap an account.
---
 src/com/android/settings/accounts/AccountTypePreferenceLoader.java | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/accounts/AccountTypePreferenceLoader.java b/src/com/android/settings/accounts/AccountTypePreferenceLoader.java
index 87aec14597..0e19dab6c0 100644
--- a/src/com/android/settings/accounts/AccountTypePreferenceLoader.java
+++ b/src/com/android/settings/accounts/AccountTypePreferenceLoader.java
@@ -34,6 +34,7 @@ import android.support.v7.preference.Preference;
 import android.support.v7.preference.Preference.OnPreferenceClickListener;
 import android.support.v7.preference.PreferenceGroup;
 import android.support.v7.preference.PreferenceScreen;
+import android.text.TextUtils;
 import android.util.Log;
 
 import com.android.settings.R;
@@ -135,8 +136,8 @@ public class AccountTypePreferenceLoader {
                 // startPreferencePanel() there. In order to inject the title string there, more
                 // dirty further hack is still needed. It's much easier and cleaner to listen to
                 // preference click event here directly.
-                if (intent.getAction().equals(
-                    android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS)) {
+                if (TextUtils.equals(intent.getAction(),
+                        android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS)) {
                     // The OnPreferenceClickListener overrides the click event completely. No intent
                     // will get fired.
                     pref.setOnPreferenceClickListener(new FragmentStarter(
-- 
2.11.0

